### PR TITLE
Auth

### PR DESCRIPTION
**Pull request by joredudiaz**
## Summary

- Auth and Unauth guards

## Jira Story

https://trello.com/c/g2NDzxDO

## Screenshot

auth

### CODE REVIEW

File: wolox-books/src/app/auth.guard.ts [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  constructor(private router: Router, private sessionService: SessionService) {  }
+
+  canActivate(): boolean {
+    if (this.sessionService.hasSession()) {
+      return true;
+    } else {
+      this.router.navigate(['login']);
+      return false;
+    }
+  }
```
Comment (FrankIglesias) [inline wolox-books/src/app/auth.guard.ts]:
```suggestion
    if (!this.sessionService.hasSession()) {
      this.router.navigate(['login']);
    } 
    return this.router.navigate(['login']);
  }
```

**Review by FrankIglesias — CHANGES_REQUESTED** [review]
Maybe you can join the logic of canActivate in one unique service and inside unauth and authguard call the same service

File: wolox-books/src/app/auth.guard.ts [hunk 2]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { SessionService } from './session.service';
+
+@Injectable({
+  providedIn: 'root'
+})
```
Comment (betoobernal) [inline wolox-books/src/app/auth.guard.ts]:
```suggestion
@Injectable()
```
You already are provided in a module.

File: wolox-books/src/app/unauth.guard.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { SessionService } from './session.service';
+
+@Injectable({
+  providedIn: 'root'
+})
```
Comment (betoobernal) [inline wolox-books/src/app/unauth.guard.ts]: same

**Comment by joredudiaz** [comment]
> Maybe you can join the logic of canActivate in one unique service and inside unauth and authguard call the same service

I think is better like this, because the code doesn't generate confusing, but if you like I do it

File: wolox-books/src/app/auth.guard.ts [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+@Injectable()
+export class AuthGuard implements CanActivate {
+  
+  constructor(private router: Router, private sessionService: SessionService) {  }
+
+  canActivate(): boolean {
+    if (!this.sessionService.hasSession()) {
+      this.router.navigate(['login']);
+    } 
+    return true
```
Comment (FrankIglesias) [inline wolox-books/src/app/auth.guard.ts]: add semi colon

File: wolox-books/src/app/unauth.guard.ts [hunk 5]
```diff
@@ -0,0 +1,17 @@
+import { Injectable } from '@angular/core';
+import { CanActivate, Router } from '@angular/router';
+import { SessionService } from './session.service';
+
+@Injectable()
+export class UnauthGuard implements CanActivate {
+  
+  constructor(private router: Router, private sessionService: SessionService) {  }
+
+  canActivate(): boolean {
```
Comment (FrankIglesias) [inline wolox-books/src/app/unauth.guard.ts:10]: Move this logic to theservice

**Review by FrankIglesias — CHANGES_REQUESTED** [review]

**Review by betoobernal — APPROVED** [review]

**Pull request merged**